tests/rptest/services/admin.py

```diff
@@ -226,34 +233,45 @@ def partition_transfer_leadership(self, namespace, topic, partition,
         path = f"partitions/{namespace}/{topic}/{partition}/transfer_leadership?target={target_id}"
         self._request("POST", path)
 
+    def get_partition_leader(self, *, namespace, topic, partition):
+        partition_info = self.get_partitions(topic=topic,
+                                             partition=partition,
+                                             namespace=namespace)
+
+        return partition_info['leader_id']
+
     def transfer_leadership_to(self, *, namespace, topic, partition, target):
```

Comment:
the changes to `transfer_leadership_to` should be a separate commit. by splitting out these changes, it also makes it easy to write a descriptive commit message subject rather than `Change admin.py`.